### PR TITLE
Feature/16348 Update pupil and school documents instead of save

### PR DESCRIPTION
**Pull request by GeorgeChatzigiannis**
Revert find methods to return lean PoJos and replaced save with update methods

### CODE REVIEW

File: admin/services/data-access/pupil.data.service.js [hunk 1]
```diff
@@ -87,10 +87,10 @@ pupilDataService.update = async function (id, doc) {
   })
 }
 
-pupilDataService.saveMultiple = async function (pupils) {
+pupilDataService.updateMultiple = async function (pupils) {
   // returns Promise
   let savedPupils = []
-  await Promise.all(pupils.map(p => p.save())).then(results => {
+  await Promise.all(pupils.map(p => Pupil.updateOne({'_id': { $eq: p._id }}, p))).then(results => {
```
Comment (GuyHarwood) [inline admin/services/data-access/pupil.data.service.js]: @GeorgeChatzigiannis can this not be an updateMany call?

File: admin/spec/service/data-access/pupil.data.service.spec.js [hunk 2]
```diff
@@ -117,21 +117,27 @@ describe('pupil.data.service', () => {
       expect(mockPupil.verify()).toBe(true)
     })
   })
-  describe('#saveMultiple', () => {
+  describe('#updateMultiple', () => {
```
Comment (GuyHarwood) [inline admin/spec/service/data-access/pupil.data.service.spec.js]: what's the hash for?

Comment (GeorgeChatzigiannis) [reply admin/spec/service/data-access/pupil.data.service.spec.js]: it's just a pattern we are using to signify that's a method.
Happy to remove it if it's confusing

File: admin/services/data-access/pupil.data.service.js (hunk 1 above)
Comment (pris54) [inline admin/services/data-access/pupil.data.service.js]: Why do you need $eq with updateOne?

**Pull request merged**